mysql-test/suite/innodb/t/lock_memory_debug.test

```diff
@@ -0,0 +1,23 @@
+--source include/have_innodb.inc
+--source include/have_debug.inc
+--source include/have_debug_sync.inc
+
+--echo #
+--echo # MDEV-34166 Server could hang with BP < 80M under stress
+--echo #
+
+call mtr.add_suppression("\\[Warning\\] InnoDB: Over 67 percent of the buffer pool.*");
+
+CREATE TABLE t1 (col1 INT) ENGINE=InnoDB;
+INSERT INTO t1 VALUES (1),(2),(3),(4),(5);
+
+SET DEBUG_DBUG="+d,innodb_skip_lock_bitmap";
+
+--error ER_LOCK_TABLE_FULL
+INSERT INTO t1 SELECT a.* FROM t1 a, t1 b, t1 c, t1 d, t1 e, t1 f, t1 g LIMIT 45000;
+
+SET DEBUG_DBUG="-d,innodb_skip_lock_bitmap";
```

Comment:
Unfortunately, the use of `-d,` will enable a "DBUG spam" mode because MariaDB does not include a related change that was made in MySQL 5.7. We should use the following pattern:
```sql
SET @save_dbug=@@debug_dbug;
SET debug_dbug='+d,innodb_skip_lock_bitmap';
# …
SET debug_dbug=@save_dbug;
```
Here, an even simpler tweak might work:
```sql
--error ER_LOCK_TABLE_FULL
SET STATEMENT debug_dbug='+d,innodb_skip_lock_bitmap' FOR
INSERT INTO t1 SELECT a.* FROM t1 a, t1 b, t1 c, t1 d, t1 e, t1 f, t1 g LIMIT 45000;
```